src/components/directory-content.jsx

```diff
@@ -413,72 +376,96 @@ const DirectoryContent = () => {
-                setElementName(childrenMetadata[element.elementUuid]?.name);
-                const subtype = childrenMetadata[element.elementUuid].subtype;
-                /** set active directory on the store because it will be used while editing the contingency name */
-                dispatch(setActiveDirectory(selectedDirectory?.elementUuid));
-                switch (element.type) {
-                    case ElementType.STUDY:
-                        let url = getLink(element.elementUuid, element.type);
-                        url
-                            ? window.open(url, '_blank')
-                            : handleError(
-                                  intl.formatMessage(
-                                      { id: 'getAppLinkError' },
-                                      { type: element.type }
-                                  )
-                              );
-                        break;
-                    case ElementType.CONTINGENCY_LIST:
-                        if (subtype === ContingencyListType.CRITERIA_BASED.id) {
-                            setCurrentFiltersContingencyListId(
-                                element.elementUuid
-                            );
-                            setOpenDialog(subtype);
-                        } else if (subtype === ContingencyListType.SCRIPT.id) {
-                            setCurrentScriptContingencyListId(
-                                element.elementUuid
-                            );
-                            setOpenDialog(subtype);
-                        } else if (
-                            subtype === ContingencyListType.EXPLICIT_NAMING.id
-                        ) {
-                            setCurrentExplicitNamingContingencyListId(
-                                element.elementUuid
-                            );
-                            setOpenDialog(subtype);
-                        }
-                        break;
-                    case ElementType.FILTER:
-                        if (subtype === FilterType.EXPLICIT_NAMING.id) {
-                            setCurrentExplicitNamingFilterId(
-                                element.elementUuid
-                            );
-                            setOpenDialog(subtype);
-                        } else if (subtype === FilterType.CRITERIA_BASED.id) {
-                            setCurrentCriteriaBasedFilterId(
-                                element.elementUuid
+            if (event.colDef.field === 'description') {
+                handleDescriptionIconClick(event);
+            } else {
+                if (childrenMetadata[event.data.elementUuid] !== undefined) {
+                    setElementName(
+                        childrenMetadata[event.data.elementUuid]?.elementName
+                    );
+                    const subtype =
+                        childrenMetadata[event.data.elementUuid]
+                            .specificMetadata.type;
```

Comment:
change from "subtype" to "type"
does that mean it was not working before ?

Reply:
How so ? 
I don't see the problem here, types are FILTER, STUDY, CONTINGENCY LIST, etc... and subtypes are mainly related to filters and contingency lists for CRITERIA_BASED, EXPLICIT_NAMING, EXPERT, etc...